crates/pypi-types/src/base_url.rs

```diff
@@ -1,37 +1,19 @@
 use serde::{Deserialize, Serialize};
 use url::Url;
 
-/// Join a possibly relative URL to a base URL.
-///
-/// When `maybe_relative` is not relative, then it is parsed and returned with
-/// `base` being ignored.
-///
-/// This is useful for parsing URLs that may be absolute or relative, with a
-/// known base URL, and that doesn't require having already parsed a `BaseUrl`.
-pub fn base_url_join_relative(base: &str, maybe_relative: &str) -> Result<Url, JoinRelativeError> {
```

Comment:
I don't think the `try` behavior is necessary here. `url::ParseError::RelativeUrlWithoutBase` occurs if and only if the URL has no scheme... And we already verify that `FileLocation::RelativeUrl` is created iff the URL has no scheme.